kuma/core/middleware.py

```diff
@@ -276,14 +287,13 @@ def process_response(self, request, response):
         return response
 
 
-class RestrictedEndpointsMiddleware(MiddlewareMixin):
+class RestrictedEndpointsMiddleware(MiddlewareBase):
+    """Restricts the accessible endpoints based on the host."""
 
-    def process_request(self, request):
-        """
-        Restricts the accessible endpoints based on the host.
-        """
+    def __call__(self, request):
         if settings.ENABLE_RESTRICTIONS_BY_HOST and is_untrusted(request):
```

Comment:
It may make sense to move this to the ``__init__``:

```
def __init__(self, get_response):
    if not settings.ENABLE_RESTRICTIONS_BY_HOST:
        raise MiddlewareNotUsed
```

Or, I may be trying too hard to use every piece of the buffalo, and it would have a negative impact on tests...

Reply:
LOL ("every piece of the buffalo") Good point! I remember reading about that new feature, and this may just be the case! I like it because if it's not ever going to be used (`settings.ENABLE_RESTRICTIONS_BY_HOST = False`), it doesn't remain embedded in the middleware stack.